Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,68 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
+    CommandsCommand.call
+  end
+
+  class CommandsCommand < Command
+    options do
+      desc "Show a list of built-in and external commands."
+      option "--quiet", "list only the names of commands without the header.", children_options: ["--include-aliases"]
```

Comment:
In the case there's child options I'd consider making `option` take a block i.e.
```ruby
option "--quiet", "list..." do
  option "--include-aliases", "the..."
end
```